### DISCUSSION TITLE
Random Quote

### DISCUSSION THREAD

**Discussion by ab-ll** [SilentVoid13/Templater#157] [Templates Showcase]
I like the daily quote function, in my daily notes.  But, I'm trying to push reviews and tasks to future dates; which creates a daily note in the future based on the template.  If I have the daily quote in the template, it will save the quote from the date that it was generated, so I will get repetitions.  The API used for the daily quote does not seem to have a parameter to select a future date so I did not see it being extensible.

Using https://github.com/lukePeavey/quotable as the background I've created the following template;

```js
<%*

var quote;
var cite;

async function updateQuote() {
    // Inspired by https://codepen.io/lukePeavey/pen/RwNVeQG
    // Fetch a random quote from the Quotable API
    //const response = await fetch("http://api.quotable.io/random?tags=famous-quotes,inspirational"); // example with tags
    const response = await fetch("http://api.quotable.io/random?tags=");
    const data = await response.json();
    if (response.ok) {
      // Update DOM elements
      quote = data.content;
      cite  = data.author;
    } else {
      quote = "An error occured";
      console.log(data);
    }
  }

await updateQuote();

tR += "> " + quote + "\r\n";
tR += "> &mdash; <cite>" + cite + "</cite>";

%>
```

Note I'm a C# guy not js, but this seems to work fine. Happy for suggestions.

**Comment by SilentVoid13** [comment]
Thanks for sharing!
